src/nbl/asset/utils/CIESProfile.cpp

```diff
@@ -0,0 +1,300 @@
+    auto outImg = asset::ICPUImage::create(std::move(imgInfo));
+
+    asset::ICPUImage::SBufferCopy region;
+    constexpr auto texelBytesz = asset::getTexelOrBlockBytesize<IES_TEXTURE_STORAGE_FORMAT>();
+    const size_t bufferRowLength = asset::IImageAssetHandlerBase::calcPitchInBlocks(width, texelBytesz);
+    region.bufferRowLength = bufferRowLength;
+    region.imageExtent = imgInfo.extent;
+    region.imageSubresource.baseArrayLayer = 0u;
+    region.imageSubresource.layerCount = 1u;
+    region.imageSubresource.mipLevel = 0u;
+    region.bufferImageHeight = 0u;
+    region.bufferOffset = 0u;
+
+    auto buffer = core::make_smart_refctd_ptr<asset::ICPUBuffer>(texelBytesz * bufferRowLength * height);
+
+    if (!outImg->setBufferAndRegions(std::move(buffer), core::make_refctd_dynamic_array<core::smart_refctd_dynamic_array<asset::IImage::SBufferCopy>>(1ull, region)))
+        return {};
+
+    //! Generate 2D IES grid data CandelaPower Distribution Curve texture can be created from
+    {
+        const auto& creationParams = outImg->getCreationParameters();
+
+        CFillImageFilter::state_type state;
+        state.outImage = outImg.get();
+        state.subresource.aspectMask = static_cast<IImage::E_ASPECT_FLAGS>(0);
+        state.subresource.baseArrayLayer = 0u;
+        state.subresource.layerCount = 1u;
+        state.outRange.extent = creationParams.extent;
+
+        const IImageFilter::IState::ColorValue::WriteMemoryInfo wInfo(creationParams.format, outImg->getBuffer()->getPointer());
+
+        const double maxValue = getMaxValue();
+        const double maxValueRecip = 1.0 / maxValue;
+
+        const double vertInv = 1.0 / height;
+        const double horiInv = 1.0 / width;
+
+        const double dTheta = core::PI<double>() / height;
+        const double dPhi = 2 * core::PI<double>() / width;
+
+        integral = 0.0;
+        size_t nonZeroEmissionDomainSize = 0u;
+
+        std::atomic<IES_STORAGE_FORMAT> aIntegral = 0.0;
+        std::atomic<size_t> aNonZeroEmissionDomainSize = 0;
+
+        auto fill = [&](uint32_t blockArrayOffset, core::vectorSIMDu32 position) -> void
+        {
+            const auto dir = octahdronUVToDir(((float)position.x + 0.5) * vertInv, ((float)position.y + 0.5) * horiInv);
+            const auto [theta, phi] = sphericalDirToRadians(dir);
+            const auto& intensity = sample(theta, phi);
+            const auto& value = intensity * maxValueRecip;
+
+            const decltype(integral) integrationV = dPhi * dTheta * std::sin(theta) * intensity;
+            
+            if (integrationV != 0.0)
+                if constexpr (policy._Parallelize)
+                {
+                    atomic_fetch_add<IES_STORAGE_FORMAT>(&aIntegral, integrationV);
+                    aNonZeroEmissionDomainSize.fetch_add(1u);
+                }
+                else
+                {
+                    integral += integrationV;
+                    ++nonZeroEmissionDomainSize;
+                }
```

Comment:
1. The integral should be performed while parsing/loading the values in (so its independent of image size)
2. While parsing the irregular grid you can work out the texture resolution needed: https://github.com/Devsh-Graphics-Programming/Nabla/pull/630#discussion_r1431351882
3. You don't just do +1 on your domain size, you need to sum `dPhi*dTheta*sin(theta)` the differential element of the domain

Do everything here while parsing